base_serializer.go

```diff
@@ -189,9 +196,12 @@ func (baseSerializer *baseSerializer) getBytes(ionType ion.Type, ionValue interf
 		return []byte{0x40}, nil
 	} else {
 		buf := bytes.Buffer{}
-		writer := NewHashWriter(ion.NewBinaryWriter(&buf), NewCryptoHasherProvider(SHA256))
+		writer, err := NewHashWriter(ion.NewBinaryWriter(&buf), NewCryptoHasherProvider(SHA256))
+		if err != nil {
+			return nil, err
+		}
 
-		err := serializers(ionType, ionValue, writer)
+		err = serializers(ionType, ionValue, writer)
 		if err != nil {
 			return nil, err
```

Comment:
It is not related to error handling; I do not know exactly the usage of the function, but do we want to return `nil` or an empty byte array here?

Reply:
I don't think the difference matters much because any calling function should be checking the returned error first before doing anything with the returned byte array.